cmd/lncli/commands.go

```diff
@@ -2259,40 +2279,73 @@ func sendPaymentRequest(ctx *cli.Context, req *lnrpc.SendRequest) error {
 		}
 	}
 
-	paymentStream, err := client.SendPayment(context.Background())
+	feeLimit, err := retrieveFeeLimit(ctx, amt)
 	if err != nil {
 		return err
 	}
+	req.FeeLimit = feeLimit
 
-	if err := paymentStream.Send(req); err != nil {
+	_, err = routerClient.SendPayment(context.Background(), req)
+	if err != nil {
 		return err
 	}
 
-	resp, err := paymentStream.Recv()
+	fmt.Printf("Payment in flight, track status by running:\n")
```

Comment:
I'm not sure we want to change the behavior of the CLI command over night like this. Instead we should add the RPC level change first, then follow up with the CLI changes. Otherwise, we may break many bash scripts that rely on `payinvoice` or `sendpayment` returning as soon as the payment is complete.

Reply:
If we don't want to break bash scripts, we'd better not change `lncli` at all. We are changing the payment result data as well. The string based "last error" isn't there anymore. No more reporting on payment attempt level outcomes.

Reply:
We're in beta, scripts will be updated. Most should probably be using the up to date payment rpc anyway.

Reply:
Have our thoughts on this matter changed?

Reply:
This is a debug commit that will be taken off before merge. I'd rather update `lncli` later.